### PR TITLE
[6.2.z] qe test coverage 1366437 + host subscriptions sub commands

### PR DESCRIPTION
**Pull request by ldjebran**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1366437 
```console
(sat-6.2.z) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/cli/test_host.py -v -k "test_positive_attache_subscription"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.32, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.2.z/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1
collected 46 items 
2017-03-30 11:26:04 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-03-30 11:26:04 - conftest - DEBUG - Collected 46 test cases


tests/foreman/cli/test_host.py::KatelloAgentTestCase::test_positive_attache_subscription PASSED

================================================= 45 tests deselected ==================================================
====================================== 1 passed, 45 deselected in 799.82 seconds =======================================
```
last test log line to show that VM is destroyed: http://pastebin.test.redhat.com/470128

### CODE REVIEW

**Comment by svtkachenko** [comment]
@ldjebran Is it possible also to cover `auto-attach`, `register` and `unregister` subcommands that mentioned in the BZ?

**Comment by ldjebran** [comment]
@svtkachenko ok will refactor this PR, create a new HostSubscriptionTestCase that will test all host subscription sub commands

**Comment by ldjebran** [comment]
```console
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.32, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.2.z/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1
collected 8 items 
2017-04-11 15:23:11 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-04-11 15:23:11 - conftest - DEBUG - Collected 8 test cases


tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_negative_without_attach PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_negative_without_attach_with_lce PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_attach PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_attach_with_lce PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_auto_attach PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_register PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_remove PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_unregister PASSED

================================================== 0 tests deselected ==================================================
============================================= 8 passed in 1538.84 seconds ==============================================
```

**Comment by ldjebran** [comment]
@JacobCallahan can please review this PR for host subscription sub commands especially the remove and auto-attach (as was not able to find requirements in docs for them), thanks

**Comment by JacobCallahan** [comment]
This looks fantastic! ACK

File: tests/foreman/cli/test_host.py [hunk 1]
```diff
@@ -1382,6 +1382,364 @@ def test_positive_register_host_ak_with_host_collection(self):
...
+            'host-id': host['id'],
+            'subscription-id': self.default_subscription_id
+        })
+        result = self._client_enable_repo()
+        self.assertEqual(result.return_code, 0)
+        # ensure that katello agent is installed
+        try:
+            self.client.install_katello_agent()
+        except VirtualMachineError as exp:
+            self.fail(u'Failed to install katello agent: {}'.format(str(exp)))
```
Comment (abalakh) [inline tests/foreman/cli/test_host.py]: self.assertNotRaises/with self.assertNotRaises? :)

**Comment by ldjebran** [comment]
@JacobCallahan many thanks

**Comment by ldjebran** [comment]
tests updated using with self.assertNotRaises
```console
(sat-6.2.z) dlezz@elysion:~/projects/robottelo-fork$ py.test -v tests/foreman/cli/test_host.py::HostSubscriptionTestCase================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.32, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.2.z/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1
collected 8 items 
2017-04-12 19:43:37 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-04-12 19:43:37 - conftest - DEBUG - Collected 8 test cases


tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_negative_without_attach PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_negative_without_attach_with_lce PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_attach PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_attach_with_lce PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_auto_attach PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_register PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_remove PASSED
tests/foreman/cli/test_host.py::HostSubscriptionTestCase::test_positive_unregister PASSED

================================================== 0 tests deselected ==================================================
============================================= 8 passed in 2005.28 seconds ==============================================
(sat-6.2.z) dlezz@elysion:~/projects/robottelo-fork$ 
```

File: robottelo/vm.py [hunk 2]
```diff
@@ -302,6 +304,8 @@ def register_contenthost(self, org, activation_key=None, lce=None,
                 username,
                 password,
             )
+            if auto_attach:
+                cmd += u' --auto-attach'
```
Comment (abalakh) [inline robottelo/vm.py]: Does auto attach work only with lce&credentials, won't it work with org+activation key?

File: tests/foreman/cli/test_host.py [hunk 3]
```diff
@@ -1382,6 +1382,358 @@ def test_positive_register_host_ak_with_host_collection(self):
...
+    def setUp(self):
+        """Create  a virtual machine without registration"""
+        super(HostSubscriptionTestCase, self).setUp()
+        self.client = VirtualMachine(distro=DISTRO_RHEL7)
+        self.client.create()
+        self.addCleanup(vm_cleanup, self.client)
+        self.client.install_katello_ca()
+
+    def _register_client(self, activation_key=None, lce=False,
+                         enable_repo=False, auto_attach=False):
```
Comment (abalakh) [inline tests/foreman/cli/test_host.py]: Please add a docstring here and in other 3 helpers.

File: tests/foreman/cli/test_host.py [hunk 4]
```diff
@@ -1382,6 +1382,358 @@ def test_positive_register_host_ak_with_host_collection(self):
...
+        self._register_client(activation_key=activation_key)
+        self.assertTrue(self.client.subscribed)
+        # attach the subscription to host
+        Host.subscription_attach({
+            'host-id': host['id'],
+            'subscription-id': self.default_subscription_id
+        })
+        result = self._client_enable_repo()
+        self.assertEqual(result.return_code, 0)
+        # ensure that katello agent is installed
```
Comment (abalakh) [inline tests/foreman/cli/test_host.py]: nitpick: /s/is/can be/

File: robottelo/vm.py (hunk 2 above)
Comment (ldjebran) [inline robottelo/vm.py]: No it does not work with with org+activation key, subscription-manager will raise an exception
when using org+activation key you can use command:
```console
subscription-manager attach --auto
```

Comment (abalakh) [inline robottelo/vm.py]: Thanks, wanted to double check just in case.

**Pull request merged**